site/content/3.12/data-science/graphml/arangoml-self-managed.md

```diff
@@ -0,0 +1,258 @@
+is ready, you can start using the ArangoML functionalities.
+
+## Node classification
+
+The following is a guide to show how to use the `arangoml` package in order to:
+- Manage projects
+- Featurize data
+- Submit training jobs
+- Evaluate model metrics
+- Generate predictions
+
+The `arangoml` package requires properly formed **specifications**. The
+specifications describe the task being performed and the data being used.
+The ArangoML services work closely together, with one task providing inputs
+to another.
+
+## Dataset: Open Intelligence GDELT
+
+This example shows how to use ArangoML to predict the **class** of `Events` in
+a Knowledge Graph constructed from the [GDETL Project](https://www.gdeltproject.org/).
+
+> GDELT monitors the world's news media from nearly every corner of every country
+> in print, broadcast, and web formats, in over 100 languages, every moment of 
+> every day. [...] Put simply, the GDELT Project is a realtime open data global
+> graph over human society as seen through the eyes of the world's news media,
+> reaching deeply into local events, reaction, discourse, and emotions of the most
+> remote corners of the world in near-realtime and making all of this available
+> as an open data firehose to enable research over human society.
+
+The events used in this example range from peaceful protests to significant
+battles. The image below depicts the connections around an example event:
+
+![ArangoML Open Intelligence Dataset Sample](../../../images/ArangoML_open_intelligence_sample.png)
+
+You can also see a larger portion of this graph, showing how the events, actors,
+news sources, and locations are interconnected into a large graph.
+
+![ArangoML Open Intelligence Dataset Graph Visualization](../../../images/ArangoML_open_intelligence_visualization.png)
+
+## Initialize ArangoML
+
+```py
+from arangoml import ArangoML
+
+arangoml = ArangoML(
+    hosts="http://localhost:8529"
+    username="root",
+    password="password",
+    projects_endpoint="http://localhost:8503",
+    training_endpoint="http://localhost:8502",
+    prediction_endpoint="http://localhost:8501",
+)
+```
+
+## Load the database
+
+```py
+from arango_datasets.datasets import Datasets
+
+DATASET_NAME = "OPEN_INTELLIGENCE_ANGOLA"
+
+username = arangoml.settings.ARANGODB_USER
+password = arangoml.settings.ARANGODB_PW
+
+# Create the Database for the Dataset
+system_db = client.db("_system", username=username, password=password, verify=True)
+system_db.delete_database(DATASET_NAME, ignore_missing=True)
+system_db.create_database(DATASET_NAME)
+
+# Load the Dataset
+dataset_db = client.db(DATASET_NAME, username=username, password=password, verify=True)
+Datasets(dataset_db).load(DATASET_NAME)
+```
+
+## Create a project
+
+```py
+# Fetch the ArangoML Project (or create it if it doesn't exist)
+project = arangoml.get_project(DATASET_NAME)
+```
+
+## Featurize the dataset
+
+```py
+# Define the Featurization Spec
+featurization_spec = {
```

Comment:
What do you think about linking to https://docs.arangodb.com/3.11/data-science/graphml/getting-started-with-arangographml/#featurization here? This way we don't have to repeat the Specification Documentation

(same question applies for Training and Prediction)

Reply:
update: this is slightly related to https://github.com/arangodb/docs-hugo/pull/419#discussion_r1446115069

Disregard the idea above, we probably shouldn't have the Self Managed doc link to the ArangoGraph doc

Would something like this work?
1. `getting-started-with-arangoml-self-managed.md`
2. `getting-started-with-arangographml.md`
3. `arangoml-specifications.md`


(where 1 & 2 both link to 3)

Reply:
ccing @geenen124

Reply:
See my comment below, was thinking of something similar :)

Reply:
Not that much content in `getting-started-with-arangoml-self-managed.md` but for the sake of clear differentiation and visibility in the navigation, it can get its own page.

Reply:
Then it should work like this:
- `getting-started-with-arangoml-self-managed.md`
- `getting-started-with-arangographml.md`
- `arangoml-specifications.md`

Will apply these changes.

Reply:
Do we want to keep then the example from "self-managed"?

Reply:
@nerpaula imo it makes sense to keep the separate example from `self-managed` - since the configuration & setup for on-prem is slightly different from on ArangoGraph